packages/zipkin/src/tracer/index.js

```diff
@@ -69,6 +69,23 @@ class Tracer {
     return childId;
   }
 
+  // creates a span, timing the given callable, adding any error as a tag
+  local(operationName, callable) {
+    return this.scoped(() => {
+      const traceId = this.createChildId();
+      this.setId(traceId);
+      this.recordAnnotation(new Annotation.LocalOperationStart(operationName));
+      try {
+        return callable();
```

Comment:
Seems way more solid, good job. For the promise return, we could simply add a check if it's a promise and chain a finally

Reply:
It will have to be more complex than that. Here are 5 unit test cases:

1) synchronous call, no error -> annotate, return success value
2) synchronous call, with error -> annotate, but rethrow the error
3) async/promise call, but throws a synchronous error -> annotate, but rethrow the error
4) async/promise call, but with async error (in a `.catch()` clause) -> annotate, but still rethrow the error so it's a rejected Promise
5) async/promise call, no error -> return the Promise with the success value

There will have to be some funky if-else-try-catch-ing going on...